### PR TITLE
DP-420 Revise API Wizard

### PR DESCRIPTION
**Pull request by tomonorman**
SQL Server functionality added.
Visual Changes
Role Creation
App Creation

### CODE REVIEW

File: app/admin_components/adf-wizard/dreamfactory-wizard.js [hunk 1]
```diff
@@ -103,8 +123,118 @@ angular.module('dfWizard', ['ngRoute', 'dfApplication', 'dfUtility', 'ngCookies'
...
+
+                    // Create the Role with access of GET, all components and active, followed by creating an
+                    // app to generate the api key.
+                    scope.dataLoading = true;
+                    $('.modal-wizard').removeClass('modal-wizard');
+
+                    // First create the role. This gets a little bit chainy / callback helly, but not too bad. We will
+                    // defer the promise so that a rejection will break the promise, until the very end.
+                    $http({
+                        method: 'POST',
```
Comment (yaroslavmo) [inline app/admin_components/adf-wizard/dreamfactory-wizard.js:189]: Is there a particular reason you did not use `dfApplicationData.saveApiData('role', requestDataObj).$promise.then(` like we do on dreamfactory-roles.js?

File: app/admin_components/adf-wizard/dreamfactory-wizard.js [hunk 2]
```diff
@@ -103,8 +123,118 @@ angular.module('dfWizard', ['ngRoute', 'dfApplication', 'dfUtility', 'ngCookies'
...
+                            var requestDataObj = {
+                                params: {
+                                    api: 'app',
+                                    fields: '*',
+                                    related: 'role_by_role_id'
+                                },
+                                data: {resource: [data]}
+                            };
+
+                            return $http({
```
Comment (yaroslavmo) [inline app/admin_components/adf-wizard/dreamfactory-wizard.js:215]: Why not `dfApplicationData.saveApiData('role', requestDataObj).$promise.then(`?

File: app/admin_components/adf-wizard/views/df-wizard-progress-bar.html [hunk 3]
```diff
@@ -0,0 +1,98 @@
+<style>
```
Comment (yaroslavmo) [inline app/admin_components/adf-wizard/views/df-wizard-progress-bar.html:1]: it is usually a bad approach to use embedded styles.  is there a way to use an external stylesheet for it? or for all wizzard styles

File: app/styles/sass/styles.css [hunk 4]
```diff
@@ -7,19 +7,19 @@
 /*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */
 @import url(font-awesome.min.css);
 @import url(lato.css);
-/* line 9, ../../../bower_components/bootstrap-sass/assets/stylesheets/bootstrap/_normalize.scss */
+@media -sass-debug-info{filename{font-family:file\:\/\/\/app\/bower_components\/bootstrap-sass\/assets\/stylesheets\/bootstrap\/_normalize\.scss}line{font-family:\000039}}
```
Comment (yaroslavmo) [inline app/styles/sass/styles.css:10]: `@media -sass-debug-info` ?

File: app/admin_components/adf-wizard/dreamfactory-wizard.js [hunk 5]
```diff
@@ -103,8 +123,118 @@ angular.module('dfWizard', ['ngRoute', 'dfApplication', 'dfUtility', 'ngCookies'
...
+                        description: roleDescription,
+                        id: null,
+                        is_active: true,
+                        lookup_by_role_id: [],
+                        name: scope.wizardData.namespace,
+                        role_service_access_by_role_id: [{
+                            component: '*',
+                            filter_op: 'AND',
+                            filters: [],
+                            requestor_mask: 1,
```
[the remark this answers is not in the document]
Comment (tomonorman) [reply app/admin_components/adf-wizard/dreamfactory-wizard.js:166]: Map added

File: app/styles/sass/styles.css (hunk 4 above)
Comment (tomonorman) [reply app/styles/sass/styles.css:10]: Thansk for the stack overflow, debug set to false in the grunt file

File: app/admin_components/adf-wizard/views/df-wizard-progress-bar.html (hunk 3 above)
Comment (tomonorman) [reply app/admin_components/adf-wizard/views/df-wizard-progress-bar.html:1]: stylesheet added for all wizards

File: app/admin_components/adf-wizard/dreamfactory-wizard.js (hunk 1 above)
Comment (tomonorman) [reply app/admin_components/adf-wizard/dreamfactory-wizard.js]: As  per discussion, adding an optional parameter to call save even if API data is not loaded allows use of `dfApplicationData.saveApiData` thanks for adding that in

**Pull request merged**